### DISCUSSION TITLE
Prioritizing sales orders by delivery date

### DISCUSSION THREAD

**Discussion by djdkochasoft** [frePPLe/frepple#504] [Q&A] [answered]
Hi, As I understand the Frepple solver, it prioritizes demand by priority and then by the due date.
Is there a way to have it prioritize sales orders by due date first, and then by priority?

**Comment by jdetaeye** [comment, marked as answer]
Use some creative SQL or python code to achieve that:
- option 1: give all demands the same priority and add a few seconds to the due date based on the demand priority
- option 2: compute externally priorities 1,2,3, 4... on each demand, such that you control very strictly the sequence followed by the planning algorithm

This can be coded in a custom app: See https://frepple.org/docs/current/developer-guide/creating-an-extension-app.php#customize-the-plan-generation
